### PR TITLE
Make url as feature and bump version

### PR DESCRIPTION
**Pull request by Its-Just-Nans**
Hello,

This PR fixes:

- https://github.com/snapview/tungstenite-rs/issues/412
- https://github.com/snapview/tungstenite-rs/issues/409

### CODE REVIEW

**Review by daniel-abramov — COMMENTED** [review]
Looks good. One question though: why is the newly introduced `tests/url_feature.rs` is a copy-paste of the existing test (with the only single exception of using `Url`)? - I think this might be a bit confusing.

**Comment by Its-Just-Nans** [comment]
Yes.
I changed the test. Is that better ?

Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 0.22.0
+- Make `url::Url` as feature
+- Add Header and subprotocol support at creation
```
Comment (daniel-abramov) [inline CHANGELOG.md:3]:
```suggestion
- Make `url` optional.
- Add a builder for convenient headers and subprotocols construction.
```

File: Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ readme = "README.md"
 homepage = "https://github.com/snapview/tungstenite-rs"
 documentation = "https://docs.rs/tungstenite/0.21.0"
```
Comment (daniel-abramov) [inline Cargo.toml:10]:
```suggestion
documentation = "https://docs.rs/tungstenite/0.22.0"
```

If the version is raised, we must also update the documentation link.

**Review by daniel-abramov — CHANGES_REQUESTED** [review]
Looks good!

Just to clarify one thing: if I got your intentions right, you want us to publish a new version after merging this PR, right? (otherwise CHANGELOG should contain `Unreleased` instead of `0.22.0` and also the update in the `Cargo.toml` would also not be necessary).

AFAIR originally we wanted to first wait for https://github.com/snapview/tungstenite-rs/pull/363 first and then release a version. But if a builder pattern is useful on its own, we could make a release before #363 I think.

**Comment by Its-Just-Nans** [comment]
> Looks good!
> 
> Just to clarify one thing: if I got your intentions right, you want us to publish a new version after merging this PR, right? (otherwise CHANGELOG should contain `Unreleased` instead of `0.22.0` and also the update in the `Cargo.toml` would also not be necessary).
> 
> AFAIR originally we wanted to first wait for #363 first and then release a version. But if a builder pattern is useful on its own, we could make a release before #363 I think.

Yes, make a release version is my goal. With a new version I could (update and) merge https://github.com/rerun-io/ewebsock/pull/27 crate which is using tungstenite

For #363 I think @conorbros is a little busy at the moment because I sent him a PR (https://github.com/shotover/tungstenite-rs/pull/1) but he didn't accept it for now

A solution could be to merge https://github.com/Its-Just-Nans/tungstenite-rs/tree/master directly into tungstenite (without passing by the fork of @conorbros). But that would make @conorbros invalid his PR. I don't know if it's a good and acceptable way of doing it

**Comment by Its-Just-Nans** [comment]
I think #363 may be mergeable now (thank you @conorbros) !

So maybe we can merge #363 then this one ?

**Comment by daniel-abramov** [comment]
That makes sense. I think we could merge #363; it just requires minor updates to pass the CI.

**Pull request merged**